updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -99,6 +100,12 @@ def check_and_create_pull_request(dependency)
             requirements_to_unlock: requirements_to_unlock
           )
 
+          if updated_deps.empty?
+            return Dependabot.logger.info(
+              "No update possible for #{dependency.name} #{dependency.version}"
+            )
+          end
```

Comment:
Under what situation can this happen? A test would help

Reply:
We found an instance internally where the NuGet updater errored out and the result was an empty update set.  While we fixed the specific issue we had, there still remains the possibility that another updater could also error out and result in an empty set, and that's why we ultimately added this check as well, to prevent the pr creator from crashing.  If there is a better place for this check we can certainly move it.

Reply:
I see.

I think it makes sense to avoid crashing randomly while trying to create a PR. However, I don't think this is an expected error condition, but a situation that's pointing at a Dependabot bug somewhere.

So, instead of just logging a message, we should be raising an error if this happens. Maybe something like:

```ruby
raise "Dependabot found some dependency requirements to unlock, yet it failed to update any dependencies"
```